src/views/Info/components/InfoNav/index.tsx

```diff
@@ -32,24 +50,95 @@ const InfoNav = () => {
   }
   return (
     <NavWrapper>
-      <Box>
-        <ButtonMenu activeIndex={activeIndex} scale="sm" variant="subtle">
-          <ButtonMenuItem as={NextLinkFromReactRouter} to="/info">
-            {t('Overview')}
-          </ButtonMenuItem>
-          <ButtonMenuItem as={NextLinkFromReactRouter} to="/info/pools">
-            {t('Pools')}
-          </ButtonMenuItem>
-          <ButtonMenuItem as={NextLinkFromReactRouter} to="/info/tokens">
-            {t('Tokens')}
-          </ButtonMenuItem>
-        </ButtonMenu>
-      </Box>
+      <Flex>
+        {!account && <NetworkSwitcher activeIndex={activeIndex} />}
+        <Box>
+          <ButtonMenu activeIndex={activeIndex} scale="sm" variant="subtle">
+            <ButtonMenuItem as={NextLinkFromReactRouter} to={`/info${chainPath}`}>
+              {t('Overview')}
+            </ButtonMenuItem>
+            <ButtonMenuItem as={NextLinkFromReactRouter} to={`/info${chainPath}/pools`}>
+              {t('Pools')}
+            </ButtonMenuItem>
+            <ButtonMenuItem as={NextLinkFromReactRouter} to={`/info${chainPath}/tokens`}>
+              {t('Tokens')}
+            </ButtonMenuItem>
+          </ButtonMenu>
+        </Box>
+      </Flex>
       <Box width={['100%', '100%', '250px']}>
         <Search />
       </Box>
     </NavWrapper>
   )
 }
 
+const targetChains = chains.filter((d) => Object.values(multiChainId).includes(d.id))
+
+export const NetworkSwitcher: React.FC<{ activeIndex: number }> = ({ activeIndex }) => {
+  const { t } = useTranslation()
+  const chainName = useGetChainName()
+  const foundChain = chains.find((d) => d.id === multiChainId[chainName])
+  const symbol = foundChain?.nativeCurrency?.symbol
+  const switchNetwork = useCallback(
+    (chainPath: string) => {
+      // don't use next router here, otherwise will have some dataflow issue
+      if (activeIndex === 0) window.location.href = `/info${chainPath}`
+      if (activeIndex === 1) window.location.href = `/info${chainPath}/pools`
+      if (activeIndex === 2) window.location.href = `/info${chainPath}/tokens`
+    },
```

Comment:
what dataflow issue we are facing?

Reply:
token list and pool list will be error
because of the block number error
for now we use this first, and I will try to fix it and useRouter later